addons/account/models/chart_template.py

```diff
@@ -948,14 +948,132 @@ def name_get(self):
+        for company in companies:
+            default_company_taxes = company.account_sale_tax_id + company.account_purchase_tax_id
+            rep_lines_accounts = templates_to_instantiate._generate_tax(company)['account_dict']
+
+            new_accounts_map = {}
+
+            # Handle tax repartition line accounts
+            tax_rep_lines_accounts_dict = rep_lines_accounts['account.tax.repartition.line']
+            for tax_rep_line, account_dict in tax_rep_lines_accounts_dict.items():
+                account_template = account_dict['account_id']
+                rep_account = new_accounts_map.get(account_template)
+
+                if not rep_account:
+
+                    existing_account = get_existing_tax_account(tax_rep_line, force_tax=default_company_taxes)
+
+                    if not existing_account:
+                        # If the default taxes were not enough to provide the account
+                        # we need, search on all other taxes.
+                        existing_account = get_existing_tax_account(tax_rep_line)
+
+                    if existing_account:
+                        rep_account = create_foreign_tax_account(existing_account, _("Foreign tax account (%s)", country.code))
+                        new_accounts_map[account_template] = rep_account
+
+                tax_rep_line.account_id = rep_account
+
+            # Handle cash basis taxes transtion account
+            caba_transition_dict = rep_lines_accounts['account.tax']
+            for tax, account_dict in caba_transition_dict.items():
+                transition_account_template = account_dict['cash_basis_transition_account_id']
+
+                if transition_account_template:
+                    transition_account = new_accounts_map.get(transition_account_template)
+
+                    if not transition_account:
+                        rep_lines = tax.invoice_repartition_line_ids + tax.refund_repartition_line_ids
+                        tax_accounts = rep_lines.account_id
+
+                        if tax_accounts:
+                            transition_account = create_foreign_tax_account(tax_accounts[0], _("Cash basis transition account"))
+
+                    tax.cash_basis_transition_account_id = transition_account
+
+            # Setup tax closing accounts on foreign tax groups ; we don't want to use the domestic accounts
+            groups = self.env['account.tax.group'].search([('country_id', '=', country.id)])
+            group_property_fields = [
+                'property_tax_payable_account_id',
+                'property_tax_receivable_account_id',
+                'property_advance_tax_payment_account_id'
+            ]
+
+            for property_field in group_property_fields:
+                default_acc = self.env['ir.property'].with_company(company)._get(property_field, 'account.tax.group')
+                if default_acc:
+                    groups.with_company(company).write({
```

Comment:
minimize the number of `with_company(company)` to avoid a cache overhead orm side:
`self.env['ir.property'].with_company(company)` & `groups.with_company(company)` can be done outside the loop